src/actions/translate.js

```diff
@@ -0,0 +1,126 @@
+const { pgClient } = require('../support/config');
+const { localeMap, pad } = require('../support/utils');
+const { load, getExhibitionPageUrlnames } = require('./load');
+
+const help = () => {
+  pad.log('Usage: npm run exhibition translate [urlname]');
+};
+
+const translate = async(page) => {
+  pad.log(`${page.urlname}:`);
+
+  if (page['other_language_codes'].length === 0) {
+    pad.log('- [no translations found]');
+  }
+
+  for (const otherLanguage of page['other_language_codes']) {
+    await translateTo(page, otherLanguage);
+  }
+
+  return page;
+};
+
+const translateTo = async(page, toLanguageCode) => {
+  const fromLocale = localeMap[page['language_code']];
+  const toLocale = localeMap[toLanguageCode];
+
+  const translatedPage = await load(page.urlname, toLanguageCode);
+
+  try {
+    pageTranslationAligned(page, translatedPage);
+    pad.log(`- ${page['language_code']} => ${toLocale}: ✔`);
+  } catch (e) {
+    pad.log(`- ${page['language_code']} => ${toLocale}: ✘ ${e.message}`);
+    return;
+  }
+
+  if (page['meta_description'][fromLocale] !== translatedPage['meta_description'][toLocale]) {
+    page['meta_description'][toLocale] = translatedPage['meta_description'][toLocale];
+  }
+
+  for (let elementIndex = 0; elementIndex < page.elements.length; elementIndex++) {
+    const element = page.elements[elementIndex];
+    const translatedElement = translatedPage.elements[elementIndex];
+
+    for (let essenceIndex = 0; essenceIndex < element.essences.length; essenceIndex++) {
+      const essence = element.essences[essenceIndex];
+
+      const translatedEssence = translatedElement.essences[essenceIndex];
+
+      for (const name in essence.data) {
+        if (!essence.data[name][fromLocale]) essence.data[name][fromLocale] = '';
+        if (translatedEssence.data[name][toLocale] && (essence.data[name][fromLocale] !== translatedEssence.data[name][toLocale]))
+          essence.data[name][toLocale] = translatedEssence.data[name][toLocale];
+      }
+    }
+  }
+
+  return page;
```

Comment:
this returns the page but actually the translations happen on the nested object(s) so this return isn't really necessary.

Reply:
But does no harm.